lib/inertia_rails/lazy.rb

```diff
@@ -0,0 +1,23 @@
+class InertiaRails::Lazy
+  def initialize(value = nil, &block)
+    @value = value
+    @block = block
+  end
+
+  def call
+    to_proc.call
+  end
+
+  def to_proc
+    # copy to local variables so they are in scope for controller.instance_exec
```

Comment:
I don't understand the need for this.

Reply:
i added a more detailed comment... things get weird because this ends up being called with `controller.instance_exec`. that changes the value of `self` while the proc gets executed, and the instance variables are unavailable within them.

copying the instance variables to local variables in the method keeps them available to the Proc via closure.